adding linters to autoware_localization_srvs

nik-tier4 opened the pull request:
`clang-tidy` giving errors. Not sure if they are relevant.

```
Error while processing /home/nithilan/AutowareArchitectureProposal/src/autoware/pilot.auto/localization/util/autoware_localization_srvs/srv/PoseWithCovarianceStamped.srv.
/home/nithilan/AutowareArchitectureProposal/src/autoware/pilot.auto/localization/util/autoware_localization_srvs/srv/PoseWithCovarianceStamped.srv:1:3: error: invalid preprocessing directive [clang-diagnostic-error]
# A sequence number is included to do primitive error checking
  ^
/home/nithilan/AutowareArchitectureProposal/src/autoware/pilot.auto/localization/util/autoware_localization_srvs/srv/PoseWithCovarianceStamped.srv:2:1: error: unknown type name 'uint32' [clang-diagnostic-error]
uint32 seq
^
/home/nithilan/AutowareArchitectureProposal/src/autoware/pilot.auto/localization/util/autoware_localization_srvs/srv/PoseWithCovarianceStamped.srv:2:11: error: expected ';' after top level declarator [clang-diagnostic-error]
uint32 seq
```

jilaada commented on localization/util/autoware_localization_srvs/package.xml:
```diff
@@ -17,6 +17,9 @@
   <member_of_group>rosidl_interface_packages</member_of_group>
 
+  <test_depend>ament_lint_auto</test_depend>
+  <test_depend>ament_cmake_cppcheck</test_depend>
```
Since this package doesn't have any C++, maybe this should be changed to `ament_lint_common`.

nik-tier4 replied:
@jilaada fixed it.

jilaada reviewed (commented):
LGTM tests pass, one comment:

jilaada reviewed (commented):
Looks like you might need to make some changes to the package.xml as it is not currently passing xmlint

nik-tier4 commented:
> Looks like you might need to make some changes to the package.xml as it is not currently passing xmlint. It is just changing the order of the various headers.

Ah, yes, fixed it.

jilaada commented:
> > Looks like you might need to make some changes to the package.xml as it is not currently passing xmlint. It is just changing the order of the various headers.
> 
> Ah, yes, fixed it.

Hmm it looks like nothing has changed?

nik-tier4 commented:
The

> > > Looks like you might need to make some changes to the package.xml as it is not currently passing xmlint. It is just changing the order of the various headers.
> > 
> > 
> > Ah, yes, fixed it.
> 
> Hmm it looks like nothing has changed?

It was complaining about the placement of these two lines:

```
  <test_depend>ament_lint_auto</test_depend>
  <test_depend>ament_lint_common</test_depend>
```
So I moved them above `<member_of_group>`

jilaada commented:
> > Hmm it looks like nothing has changed?
> 
> It was complaining about the placement of these two lines:
> 
> ```
>   <test_depend>ament_lint_auto</test_depend>
>   <test_depend>ament_lint_common</test_depend>
> ```
> 
> So I moved them above `<member_of_group>`

Ahh sorry, the diff didn't show that clearly because of the force push! Thanks for making those changes :+1:
